homework/list.js

```diff
@@ -1,79 +1,83 @@
 'use strict';
 
-(function() {
+(function(exports) {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
-
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
+  var ListManager = function(){
+  	this._listNoteContent = [];
+	this._wrapper = document.querySelector('#note-list-wrapper');
   }
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+  ListManager.prototype = {
+	start(){
+		this.fetchList().then((function(data){
+			this.updateList(data);
+			this.drawList();
+			this.preloadFirstNote();
+		}).bind(this)).catch((function(){
```

Comment:
**Serious issue**: if you catch it without **logging AND re-throw** it, the following chains (if any) will not get interrupted and in the console there will be nothing about the error. It could be a disaster in real programs. So please correct it.